src/headers.rs

```diff
@@ -1,17 +1,78 @@
-pub enum Header {
+use http::request::Builder;
+use std::collections::HashMap;
+
+pub enum HeaderField {
     User(String),
     Source(String),
     Catalog(String),
     Schema(String),
     // TimeZone,  // Not yet supported
     // Language,  // Not yet supported
     TraceToken(String),
-    Session(String, String),
+    Session(HashMap<String, String>),
```

Comment:
I've been going back and forth whether it's a better API to have people supply multiple `Session(String, String)`, or one `Session(HashMap<String, String>)`.

The main differences is that map literals are a bit awkward in Rust, although the `maplit` crate makes them easy.